byzcoin/tx_pipeline.go

```diff
@@ -110,7 +110,11 @@ func (s *defaultTxProcessor) CollectTx() (*collectTxResult, error) {
 		return nil, xerrors.Errorf("reading config: %v", err)
 	}
 
-	_, isNotProcessingBlock := s.skService().WaitBlock(s.scID, nil)
+	if s.skService().ChainIsProcessing(s.scID) {
+		// When a block is processed, wait for the block to be done
+		time.Sleep(bcConfig.BlockInterval / 2)
+		return &collectTxResult{Txs: nil, CommonVersion: CurrentVersion - 1}, nil
```

Comment:
That looks weird, you wait for some time then return the same data you'd have before sleeping.

Reply:
Looking at the call chain, I think I can remove that one. Sending it to Travis for confirmation ;)